modules/imgproc/src/featureselect.cpp

```diff
@@ -54,7 +54,8 @@ struct greaterThanPtr :
         public std::binary_function<const float *, const float *, bool>
 {
     bool operator () (const float * a, const float * b) const
-    { return *a > *b; }
+    // Ensure a fully deterministic result of the sort
+    { return (*a > *b) ? true : (*a < *b) ? false : (a > b); }
```

Comment:
The addresses 'a' and 'b' are always of pixels within the same Mat, so their ordering is deterministic.  If you look at the rest of the GFFT code you will see that it uses the pixel addresses as a shorthand rather than explicitly storing the x,y coords (the way that the OCL version below does).